recurring_contract/models/move.py

```diff
@@ -178,8 +177,101 @@ def find_sum(numbers, target, partial=None):
                     if payment_line.credit > missing_amount:
                         # Split last added line amount to perfectly match
                         # the total amount we are looking for
-                        return (open_payments[:index + 1] | move_lines)\
+                        return (open_payments[:index + 1] | move_lines) \
                             .split_payment_and_reconcile()
                     payment_amount += payment_line.credit
                 return (open_payments | move_lines).reconcile()
 
+    def update_invoices(self, updt_val):
+        """
+        It updates the invoices in self with the value of updt_val
+
+        :param updt_val: a dictionary of invoices values with the invoice name
+        which refer to another dictionnary of values for that invoice name
+        """
+        for invoice in self:
+            invoice.button_draft()
+            # Retrieve the value for a specific invoice
+            val_to_updt = updt_val.get(invoice.name)
+            # If the invoice is defined to_cancel we update it state
+            if "to_cancel" in val_to_updt["invoice_line_ids"]:
+                invoice._cancel_invoice()
+                continue
+            else:
+                invoice.write(val_to_updt)
+            invoice.action_post()
+
+    def _build_invoice_data(self, inv_lines=False, due_date=False, ref=False, pay_mode_id=False):
+        """
+        It takes a list of invoice lines, a due date, a payment reference, and a payment mode, and returns a dictionary with
+        the invoice lines, the due date, the payment reference, and the payment mode
+
+        :param inv_lines: a list of dictionaries containing the invoice lines to be added to the invoice, defaults to False
+        (optional)
+        :param due_date: The date at which the invoice should be paid, defaults to False (optional)
+        :param ref: The reference of the invoice, defaults to False (optional)
+        :param pay_mode_id: The payment mode to be used for the invoice, defaults to False (optional)
+        :return: A dictionary with the invoice_line_ids, date, payment_reference, and payment_mode_id.
+        """
+        # Ensure that the function receive only one invoice
+        self.ensure_one()
+        # Build the dictionnary
+        inv_val_dict = dict()
+        inv_val_dict[self.name] = {
+            'invoice_line_ids': self._build_invoice_lines(inv_lines) if inv_lines else self.invoice_line_ids.ids,
+            'date': due_date if due_date else self.date,
+            'payment_reference': ref if ref else self.payment_reference,
+            'payment_mode_id': pay_mode_id if pay_mode_id else self.payment_mode_id.id
+        }
+        return inv_val_dict
```

Comment:
All the else parts, if no value are provided, will simply write the same value as before. Isn't it better to not include it in the list of change in that case?
```suggestion
        inv_val_dict = {}
        if inv_lines:
            inv_val_dict["invoice_line_ids"] = inv_lines
        if due_date:
            inv_val_dict["date"] = due_date
        if ref:
            inv_val_dict["payment_reference"] = ref
        if pay_mode_id:
            inv_val_dict["payment_mode_id"] = pay_mode_id
        return {self.name: inv_val_dict}
```